protocols/secio/src/codec/decode.rs

```diff
@@ -42,21 +43,23 @@ pub struct DecoderMiddleware<S> {
     cipher_state: StreamCipher,
     hmac: Hmac,
     raw_stream: S,
+    nonce: Vec<u8>
 }
 
 impl<S> DecoderMiddleware<S> {
     #[inline]
-    pub fn new(
-        raw_stream: S,
-        cipher: StreamCipher,
-        hmac: Hmac,
-    ) -> DecoderMiddleware<S> {
+    pub fn new(raw_stream: S, cipher: StreamCipher, hmac: Hmac) -> DecoderMiddleware<S> {
```

Comment:
Why not put the `nonce` as part of the constructor? I don't see why we would change it later.

Reply:
Also some documentation saying that `nonce` is what the stream of data has to start with would be nice.